tests/target/issue-3669.rs

```diff
@@ -0,0 +1,77 @@
+pub trait PCG:
+    self::sealed::Sealed // comment1
+    + Sized                // comment2
+    + Eq                   // comment3
+    + Hash                 // comment4
+    + Debug                // comment5
+    + Clone                // comment6
+    + Default              // comment7
+    + Serialize            // comment8
+    + for<'a> Deserialize<'a> // comment9
+{
+    type DoubleState: Copy                                // Note(Evrey): Because Rust is drunk. 1
+        + ShrAssign<u8>                       // Note(Evrey): Because Rust is drunk. 2
+        + Shl<u8, Output = Self::DoubleState> // Note(Evrey): Because Rust is drunk. 3
+        + BitAnd<Output = Self::DoubleState>  // Note(Evrey): Because Rust is drunk. 4
+        + BitOrAssign                         // Note(Evrey): Because Rust is drunk. 5
+        + Sub<Output = Self::DoubleState>     // Note(Evrey): Because Rust is drunk. 6
+        + Into<u128>                          // Note(Evrey): Because Rust is drunk. 7
+        + Debug                               // Note(Evrey): Because Rust is drunk. 8
+        + Eq                                  // Note(Evrey): Because Rust is drunk. 9
+        + Hash                                // Note(Evrey): Because Rust is drunk. 10
+        + Default                             // Note(Evrey): Because Rust is drunk. 11
+        + Serialize                           // Note(Evrey): Because Rust is drunk. 12
+        + for<'a> Deserialize<'a>; // Note(Evrey): Because Rust is drunk. 13
+}
+
+pub trait Bar:
+    self::sealed::Sealed // comment1
+    + Sized                // comment2
+    + Eq                   // comment3
+    + Hash                 // comment4
+    + Debug                // comment5
+    + Clone                // comment6
+    + Default              // comment7
+    + Serialize            // comment8
+    + for<'a> Deserialize<'a> // comment9
+{
+    type DoubleState: Copy                                // Note(Evrey): Because Rust is drunk. 1
```

Comment:
Looks like the first comment is not aligned with others. Is this intentional?

Reply:
the alignement in https://github.com/rust-lang/rustfmt/pull/3680/commits/3850fb1479bf5a1082d25d25e59066f9639b167b#diff-11ffc82e7dc3c14ae179a08592fd9c26R28 got fixed but not this one yet, i'll try to see if I can.

It seems that the shape offset is not correctly set: the extra space taken by the type name is not considered by the first item.